src/Authentication/Authentication/test/Find-MgGraphPermission.Tests.ps1

```diff
@@ -0,0 +1,417 @@
+            { Find-MgGraphPermission 'ReadWrite' | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest -Exactly 1
+            { Find-MgGraphPermission 'Email' | Out-Null }
+            Assert-MockCalled Invoke-MgGraphRequest -Exactly 1
+        }
+
+        It 'Calls Invoke-MgGraphRequest for the first request and then only if the Online parameter is specified' {
+            Find-MgGraphPermission 'role' | Should -Not -Be $null
+            Assert-MockCalled Invoke-MgGraphRequest -Exactly 1
+            Find-MgGraphPermission 'user' | Should -Not -Be $null
+            Assert-MockCalled Invoke-MgGraphRequest -Exactly 1
+            Find-MgGraphPermission 'user' -Online | Should -Not -Be $null
+            Assert-MockCalled Invoke-MgGraphRequest -Exactly 2
+        }
+
+        It "Retrieves the expected set of delegated and app-only permissions when a search string is specified" {
+            { Find-MgGraphPermission 'ReadWrite' | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+            $test = Find-MgGraphPermission 'ReadWrite'
+            $test.GetType() | Should -Be 'System.Object[]'
+            $test.length | Should -Be 4
+            $test.PermissionType[0] | Should -Be 'Delegated'
+            (_Permissions_State).msGraphServicePrincipal | Should -Not -Be $null
+            (_Permissions_State).isFromInvokeMgGraphRequest | Should -Be $true
+            { Find-MgGraphPermission 'ReadWrite' -Online | Out-Null } | Should -Not -Throw
+            $testOnline = Find-MgGraphPermission 'ReadWrite' -Online
+            $testOnline.length | Should -Be 4
+        }
+
+        It "Retrieves only delegated permissions even the PermissionType parameter is specified as Delegated when application permissions would also match the search string" {
+            $allMatches = Find-MgGraphPermission ReadWrite
+
+            $index = 0
+
+            $delegatedPermissions = , 'RoleAssignmentSchedule.ReadWrite.Directory'
+            $applicationPermissions = 'Directory.ReadWrite.All', 'Group.ReadWrite.All', 'WorkforceIntegration.ReadWrite.All'
+
+            $allPermissions = @()
+            $allPermissions += $delegatedPermissions
+            $allPermissions += $applicationPermissions
+
+            $allPermissions | ForEach-Object {
+                $allMatches[$index++].Name | Should -Be $_
+            }
+
+            $delegatedMatches = Find-MgGraphPermission ReadWrite -PermissionType Delegated
+
+            $delegatedIndex = 0
+
+            $delegatedPermissions | ForEach-Object {
+                $delegatedMatches[$delegatedIndex++].Name | Should -Be $_
+            }
+        }
+
+        It "Should return null and not throw an exception if ExactMatch is specified and there is no match" {
+            { Find-MgGraphPermission -ExactMatch IDontExist 2>&1 | out-null } | Should -Not -Throw
+            $ErrorActionPreference = 'continue'
```

Comment:
`$ErrorActionPreference = 'continue'` should be specified before making the call. This is the cause of https://github.com/microsoftgraph/msgraph-sdk-powershell/pull/809#issuecomment-895325576.
```suggestion
            $ErrorActionPreference = 'continue'
            { Find-MgGraphPermission -ExactMatch IDontExist 2>&1 | out-null } | Should -Not -Throw
```

Reply:
> `$ErrorActionPreference = 'continue'` should be specified before making the call. This is the cause of [#809 (comment)](https://github.com/microsoftgraph/msgraph-sdk-powershell/pull/809#issuecomment-895325576).

Yeah, we ended up doing that @peombwa as a workaround. But technically this is incorrect -- the default value for `$ErrorActionPreference` is `Continue` as stated in the [documentation](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_preference_variables?view=powershell-7.1#erroractionpreference). As far as validating the functionality our workaround will get the job done (maybe even passing that erroraction preference as a parameter instead of using the variable), but it seems like something in the test infrastructure is altering the standard powershell preferences.